src/common/filesystembase.cpp

```diff
@@ -111,36 +111,42 @@ static QFile::Permissions getDefaultWritePermissions()
 void FileSystem::setFileReadOnly(const QString &filename, bool readonly)
 {
 #ifdef  Q_OS_WIN
-    if (isLnkFile(filename)) {
-        if (!fileExists(filename)) {
-            return;
-        }
-        try {
-            const auto permissions = filePermissionsWin(filename);
+    if (!fileExists(filename)) {
+        Q_ASSERT(false);
+        return;
+    }
 
-            std::filesystem::perms allWritePermissions = std::filesystem::perms::_All_write;
-            static std::filesystem::perms defaultWritePermissions = std::filesystem::perms::others_write;
+    const auto fileAttributes = GetFileAttributesW(filename.toStdWString().c_str());
+    if (fileAttributes == INVALID_FILE_ATTRIBUTES) {
+        const auto lastError = GetLastError();
+        auto errorMessage = static_cast<char*>(nullptr);
+        if (FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                          nullptr, lastError, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), errorMessage, 0, nullptr) == 0) {
+            qCWarning(lcFileSystem()) << "GetFileAttributesW" << filename << (readonly ? "readonly" : "read write") << errorMessage;
+        } else {
+            qCWarning(lcFileSystem()) << "GetFileAttributesW" << filename << (readonly ? "readonly" : "read write") << "unknown error" << lastError;
+        }
+    }
```

Comment:
I don't think it makes sense to attempt setting the file attributes when it turned out to be `INVALID_FILE_ATTRIBUTES`

```suggestion
       return;
    }
```

Reply:
makes sense
done